### PR TITLE
Added PEFT docs

### PR DESCRIPTION
**Pull request by merveenoyan**

### CODE REVIEW

**Comment by merveenoyan** [comment]
Adding light screenshots.

File: docs/hub/peft.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Using PEFT at Hugging Face
+
+🤗 [Parameter-Efficient Fine-Tuning (PEFT)](https://huggingface.co/docs/peft/index), is a library for efficiently adapting pre-trained language models to various downstream applications without fine-tuning all the model’s parameters.
```
Comment (osanseviero) [inline docs/hub/peft.md:3]:
```suggestion
🤗 [Parameter-Efficient Fine-Tuning (PEFT)](https://huggingface.co/docs/peft/index) is a library for efficiently adapting pre-trained language models to various downstream applications without fine-tuning all the model’s parameters. 
```

File: docs/hub/peft.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# Using PEFT at Hugging Face
```
Comment (osanseviero) [inline docs/hub/peft.md:1]: Let's also add to https://huggingface.co/docs/hub/models-libraries

File: docs/hub/peft.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+All PEFT models can be loaded from Hub. Loading PEFT models require loading base models that were fine-tuned, so we need to load the transformers model as well.
+
+```py
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from peft import PeftModel, PeftConfig
+
+config = PeftConfig.from_pretrained("dfurman/Mistral-7B-Instruct-v0.2")
+model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-v0.1")
+model = PeftModel.from_pretrained(model, "dfurman/Mistral-7B-Instruct-v0.2")
+tokenizer = AutoTokenizer.from_pretrained(config.base_model_name_or_path)
```
Comment (osanseviero) [inline docs/hub/peft.md:30]: Maybe this makes things a bit cleaner
```suggestion
base_model = "mistralai/Mistral-7B-v0.1"
adapter_model = "dfurman/Mistral-7B-Instruct-v0.2"

model = AutoModelForCausalLM.from_pretrained(base_model)
model = PeftModel.from_pretrained(model, adapter_model)
tokenizer = AutoTokenizer.from_pretrained(base_model)
```

File: docs/hub/peft.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    outputs = model.generate(input_ids=inputs["input_ids"].to("cuda"), max_new_tokens=10)
+    print(tokenizer.batch_decode(outputs.detach().cpu().numpy(), skip_special_tokens=True)[0])
+```
+
+It outputs the following:
+
+```text
+Tell me the recipe for chocolate chip cookie dough.
+
+1. Preheat oven
```
Comment (osanseviero) [inline docs/hub/peft.md:51]: Maybe increase the number of generated tokens so we have a nicer example

**Review by osanseviero — COMMENTED** [review]
Cool!

File: docs/hub/peft.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+🤗 [Parameter-Efficient Fine-Tuning (PEFT)](https://huggingface.co/docs/peft/index) is a library for efficiently adapting pre-trained language models to various downstream applications without fine-tuning all the model’s parameters. 
+
+## Exploring PEFT on the Hub
+
+You can find PEFT models by filtering at the left of the [models page](https://huggingface.co/models?library=peft&sort=trending).
+
+
+## Installation
+
+To get started, you can check out the [Quick Tour of PEFT docs](https://huggingface.co/docs/peft/main/en/quicktour#quicktour). To install, follow the [PEFT installation guide](https://huggingface.co/docs/peft/install).
```
Comment (pcuenca) [inline docs/hub/peft.md:12]:
```suggestion
To get started, you can check out the [Quick Tour in the PEFT docs](https://huggingface.co/docs/peft/quicktour). To install, follow the [PEFT installation guide](https://huggingface.co/docs/peft/install).
```

File: docs/hub/peft.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+You can also use the following one-line install through pip:
+
+```
+$ pip install peft
+```
+
+## Using existing models
+
+All PEFT models can be loaded from Hub. Loading PEFT models require loading base models that were fine-tuned, so we need to load the transformers model as well.
```
Comment (pcuenca) [inline docs/hub/peft.md:21]:
```suggestion
All PEFT models can be loaded from the Hub. To use a PEFT model you also need to load the base model that was fine-tuned, as shown below.
```

File: docs/hub/peft.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+model = AutoModelForCausalLM.from_pretrained(base_model)
+model = PeftModel.from_pretrained(model, adapter_model)
+tokenizer = AutoTokenizer.from_pretrained(base_model)
+
+model = model.to("cuda")
+model.eval()
+```
+
+Once loaded, you can pass inputs to tokenizer to prepare them, and call `model.generate()` in regular PyTorch fashion.
```
Comment (pcuenca) [inline docs/hub/peft.md:38]:
```suggestion
Once loaded, you can pass your inputs to the tokenizer to prepare them, and call `model.generate()` in regular `transformers` fashion.
```

File: docs/hub/peft.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+</div>
+<div class="flex justify-center">
+<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/peft_snippet_light.png"/>
+<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/peft_snippet.png"/>
+</div>
+
+## Additional resources
+
+* PEFT [repository](https://github.com/huggingface/peft)
+* PEFT [docs](https://huggingface.co/docs/peft/index)
```
Comment (pcuenca) [inline docs/hub/peft.md:76]:
```suggestion
* PEFT [docs](https://huggingface.co/docs/peft/index)
* PEFT [models](https://huggingface.co/models?library=peft&sort=trending)
```

**Review by pcuenca — APPROVED** [review]
Very nice! What do you think about mentioning that the base model is documented in the model card of the PEFT fine-tune? (for example: https://huggingface.co/dfurman/Mistral-7B-Instruct-v0.2/blob/6baefb2de501f41cd1897cb661b2cd474a3e9961/README.md?code=true#L12). Not sure it's necessary.
